op-chain-ops/cmd/check-ecotone/main.go

```diff
@@ -262,12 +340,25 @@ func check4844Precompile(ctx context.Context, env *actionEnv) error {
 			continue
 		}
 		env.log.Info("Beacon block root query timestamp", "query_timestamp", ti)
-		// revert when timestamp doesn't exist (when not aligned it won't exist)
+		// revert when timestamp doesn't exist (when not aligned it won't exist),
+		// or when we call it at the activation block and the contract was newly deployed
+		// (the beacon block root is processed at the start of the block,
+		// but the contract might not exist yet, during activation).
+		revert := ti%conf.BlockTime != alignment
+		if conf.IsEcotoneActivationBlock(ti) {
+			// if the contract already existed, then we deployed the nonce=1 contract during upgrade
+			code, err := env.l2.CodeAt(ctx, crypto.CreateAddress(derive.EIP4788From, 1), nil)
+			if err != nil {
+				return fmt.Errorf("failed to check code: %w", err)
+			}
+			revert = revert || len(code) > 0
```

Comment:
```suggestion
			revert = revert || len(code) == 0
```

Correction: when the nonce=1 contract doesn't exist, then the nonce=0 didn't exist during upgrade, and thus reverts.

This isn't covered when already way past the upgrade, but happens if running the check at or closely after the upgrade happens.